### PR TITLE
adding Obsidian HackerNews

### PR DESCRIPTION
**Pull request by arpitbbhayani**
Periodically fetches and displays top stories from HackerNews.

# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/arpitbbhayani/obsidian-hackernews

## Release Checklist

- [x] I have tested this on Windows, macOS, and Linux _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] README clearly describes the plugins purpose and provides clear usage instructions.
- [x] I have added a license in the LICENSE file.

### CODE REVIEW

**Comment by lishid** [comment]
https://github.com/arpitbbhayani/obsidian-hackernews/blob/d03d91fe86fa59120d43483b2b30cfdb00116806/src/main.ts#L20
This `Promise.all` doesn't seem necessary.

https://github.com/arpitbbhayani/obsidian-hackernews/blob/d03d91fe86fa59120d43483b2b30cfdb00116806/src/main.ts#L53
Usually the procedure here is to `for(let leaf of getLeavesOfType()) { leaf.detach() }`.

https://github.com/arpitbbhayani/obsidian-hackernews/blob/d03d91fe86fa59120d43483b2b30cfdb00116806/src/apiManager.ts#L3
Should import `moment` from `obsidian` so you don't package another copy in your plugin.

https://github.com/arpitbbhayani/obsidian-hackernews/blob/d03d91fe86fa59120d43483b2b30cfdb00116806/src/apiManager.ts#L2
Don't use `path` since it's only available on Electron. Instead, just join with `/` and then use `normalizePath` from `obsidian`.

https://github.com/arpitbbhayani/obsidian-hackernews/blob/d03d91fe86fa59120d43483b2b30cfdb00116806/src/apiManager.ts#L38

Could probably put vault in a variable here to avoid the duplicated chain.

**Comment by arpitbbhayani** [comment]
Thank you @lishid for reviewing. I have released v 1.0.4 incorporating all the mentioned changes

 - removed unnecessary `Promise.all`
 - detaching all the leaves instead of just one
 - using Obsidian's `moment` instead of importing the Moment package
 - Have joined the directory and filename instead of using `path`
 - have put `vault` in a separate variable to avoid duplicated chains.

I have also tested the code through various flows and everything is working fine.

**Comment by lishid** [comment]
https://github.com/arpitbbhayani/obsidian-hackernews/blob/2c242de3c92cf0f70b7ada93ad275115242d7672/src/apiManager.ts#L2
While this works, you should probably just add `moment` to the import list above.
